cmd/kubeadm/app/util/apiclient/wait.go

```diff
@@ -236,6 +243,15 @@ func (w *KubeWaiter) WaitForStaticPodHashChange(nodeName, component, previousHas
 
 		return true, nil
 	})
+
+	// distinguish getStaticPodSingleHash err and kubelet possible err here
+	if err != nil {
+		if lastErr != nil {
+			return errors.Wrapf(lastErr, "failed to obtain static Pod hash for component %s on Node %s", component, nodeName)
+		}
+		return errors.Wrapf(err, "static pod hash didn't change for a while, kubelet may fail to restart it")
+	}
+	return nil
```

Comment:
```suggestion
	// If the error is a timeout for unchanged hash, return a more specific error
	if lastErr != nil {
		return errors.Wrapf(lastErr, "failed to obtain static Pod hash for component %s on Node %s", component, nodeName)
	}
	if err != nil {
		return errors.Wrapf(err, "static pod hash didn't change for a while, kubelet may fail to restart it")
	}
	return nil
```

Nit: to make less nested `if-else`.

Reply:
Yes, thanks, it makes sense.
But we both ignored one point that if `getStaticPodSingleHash` return an err in the first time and return no err after some retries, `lastErr` was still left not nil, if we directly check `lastErr` it will return wrong err becasue it maybe has already no err or be anoher err. So I add `lastErr = nil` when `getStaticPodSingleHash` has returned  no err  after some retries, which can distinguish different errs.